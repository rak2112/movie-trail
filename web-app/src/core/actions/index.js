export * from './api';
export * from './entities';
export * from './movie-details';
export * from './user';