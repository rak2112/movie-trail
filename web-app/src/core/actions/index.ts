export * from './api';
export * from './entities';
export * from './movie-details';
export * from './search-movies';
export * from './user';