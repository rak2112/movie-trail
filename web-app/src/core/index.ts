// // components
// export * from './components/carousel';
// export * from './components/form-input';
// export * from './components/form-validation';
// export * from './components/imageLoader';
// export * from './components/load-error';
// export * from './components/loader';
// export * from './components/loader-wrapper';
// export * from './components/movie';
// export * from './components/movie-list';
// export * from './components/pagination';
// export * from './components/toast';
// export * from './components/user-actions';

// // containers
// export {default as Error } from './containers/error';

// // utils
// export * from './utils/movies-service';
// export * from './utils/util-service';
// export * from './utils/user-service';
// export * from './utils/localStorage';
// export * from './utils/http';

// // store 
// export * from './store/configureStore';

// // // selectors
// export * from './selectors/movie-details';
// export * from './selectors/movies';


// // interfaces 
// export * from './interfaces';

// // routes
// export * from './routes';

// // reducders
// export * from './reducers';

// // actions
// export * from './actions/api';
// export * from './actions/entities';
// export * from './actions/movie-details';
// export * from './actions/user';