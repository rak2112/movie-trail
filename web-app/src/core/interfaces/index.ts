type MovieType =
  | 'Action'
  | 'Adventure'
  | 'Animation'
  | 'Comedy'
  | 'Crime'
  | 'Documentry'
  | 'Drama'
  | 'Family'
  | 'Fantasy'
  | 'History'
  | 'Horror'
  | 'Music'
  | 'Mystery'
  | 'Romance'
  | 'Science Fiction'
  | 'TV Movie'
  | 'Thriller'
  | 'War'
  | 'Western';

export type UserMovieType= 'FAVORITE' | 'WATCHLIST';

export type UserView = 'Favorites' | 'Watchlist';

export interface Api {
  isFetching?: boolean;
  hasError?: boolean;
  errorMessage?: string;
}; 

export interface FormValues {
  name?: string;
  email: string;
  password: string;
  confirmPassword?: string;
  token?: string;
}

export interface Language {
  id: string;
  name: string;
}

export interface Movie {
  id: number;
  backdrop_path: string;
  genre_ids: number[];
  genres: Genre[];
  homepage: string;
  overview: string;
  poster_path?: string;
  runtime: string;
  release_date: string;
  spoken_languages: Language[];
  status: string;
  title: string;
  type?: string;
  vote_average: string;
};

export interface Movies {
  pageNo: number;
  totalPages: number;
  totalResults: number;
  results: Movie[];
  redirection? : boolean;
};

export interface MoviesData {
  pageNo: number;
  total_results: number;
  total_pages: number;
  results?: Movie[];
};

export interface MoviesRes {
  movies: Movie[]
};

export interface Genre {
  id: number;
  name: MovieType;
};

export interface GenresMap {
  [id: number]: Genre;
};

export interface UserMovie extends Movie {
  _id?: string;
  movieType?: UserMovieType;
}

export interface User {
  email?: string;
  displayName?: string;
  loggedIn?: boolean;
  movies?: UserMovie[];
  resetMessage?: string;
  token?: string;
}

export interface UserMovies {
  [id: string]: UserMovie;
}
export interface UserMovieMap {
  favorites: UserMovies;
  watchlist: UserMovies;
}

export interface Cast {
  character: string;
  id: string;
  name: string;
  profile_path?: string;
}

export interface Crew extends Cast {
  department: string;
  job: string;
}

export interface Poster {
  file_path: string;
}

export interface Persons {
  id?: string;
  cast: Cast[];
  crew: Crew[];
}

export interface Posters {
  id?: string;
  backdrops: Poster[];
  posters: Poster[];
}

export interface Video {
  id: string;
  key: string;
  name: string;
}

export interface Videos {
  id: string;
  results: Video[]
}

export interface MovieDetail {
  isFetching?: boolean;
  hasError?: boolean;
  details: Movie;
  images: Posters;
  persons: Persons;
  videos: Videos
}