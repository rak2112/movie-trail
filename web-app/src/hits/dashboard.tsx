import React from 'react';
import Movies, { MoviesProps } from '../all-movies/dashboard.container';

const Hits = (props: MoviesProps) =>  (<Movies {...props}/>);

export default Hits;
