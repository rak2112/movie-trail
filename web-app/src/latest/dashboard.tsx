import React from 'react';
import Movies, { MoviesProps } from '../all-movies/dashboard.container';

const Latest = (props: MoviesProps) =>  (<Movies {...props} />);

export default Latest;