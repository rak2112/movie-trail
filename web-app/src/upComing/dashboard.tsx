import React from 'react';
import Movies from '../all-movies/dashboard.container';

const UpComing = (props: any) =>  (<Movies {...props} route={'up-coming'}/>);

export default UpComing;
