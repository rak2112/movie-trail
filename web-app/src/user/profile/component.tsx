import React, { PureComponent } from 'react';
import { Nav, NavItem, Tab, Tabs } from 'react-bootstrap';
import styled from 'react-emotion';
import {
  MovieList,
  Toaster 
} from '../../core/components';
import { Profile } from './container';
import { MovieTabs } from './tabs';
import { ProfileDetails } from './user-details.component';


export class ProfileComponent extends PureComponent <Profile, any> {
  static displayName = 'MovieComponent';
  render() { 
    const { user } = this.props;
    return (
      <>
        <MovieList>
          {() => (
            <>
              <ProfileDetails {...user}/>
              <Toaster duration={2000}/>  
              <MovieTabs {...this.props}/>
            </>
          )}
        </MovieList>
      </>
    )
  }
}