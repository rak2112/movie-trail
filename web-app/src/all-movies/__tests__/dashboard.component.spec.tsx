import { mount, shallow } from 'enzyme';
import React from "react";
import { StaticRouter } from 'react-router'
import { create } from 'react-test-renderer';

import { Genre, Movies, UserMovieMap } from '../../core/interfaces';
import { MoviesComponent } from '../dashboard.component';

let context: any;;
describe(`Movies Component`, () => {
  const getUuid = jest.fn();
  const props = {
    addMovie: jest.fn(),
    deleteMovie: jest.fn(),
    genres:[
      {id:2, name: "Action"} as Genre
    ], 
    loadMovies: jest.fn(),
    movies: {
      totalPages: 1, pageNo: 1, totalResults: 3, 
      results: [{
        id: 2,
        backdrop_path: 'path',
        genre_ids: [1, 2],
        genres: [],
        homepage: 'homepage',
        overview: 'overview',
        runtime: '107',
        release_date: '2-04-19',
        spoken_languages: [],
        status: 'status',
        title: 'title',
        vote_average: 'vote_avg'
      },
      {
        id: 3,
        backdrop_path: 'path',
        genre_ids: [1, 2],
        genres: [],
        homepage: 'homepage',
        overview: 'overview',
        runtime: '107',
        release_date: '2-04-19',
        spoken_languages: [],
        status: 'status',
        title: 'title',
        vote_average: 'vote_avg'
      }]
    } as Movies,
    userMovies: {
      favorites: {
        2: {
          id: 2,
          _id: '23',
          backdrop_path: 'path',
          genre_ids: [1, 2],
          genres: [],
          homepage: 'homepage',
          movieType: 'FAVORITE',
          overview: 'overview',
          runtime: '107',
          release_date: '2-04-19',
          spoken_languages: [],
          status: 'status',
          title: 'title',
          vote_average: 'vote_avg'
        }
      },
      watchlist: {}
    } as UserMovieMap
  };

  beforeEach(() => {
    context = {};
    // getUuid.mockReturnValue('uuid');
    // jest.mock('../../core/components/movie', () => 'MovieComponent');
  });

  fit(`should match snapshot`, () => {
    const component = create(
      <StaticRouter location="someLocation" context={context}>
        {/* used to wrap Router around the Link */}
        <MoviesComponent {...props}/>
      </StaticRouter>
      
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
  
  it(`should have children`, () => {
    const wrapper = mount(
      <StaticRouter location="someLocation" context={context}>
        <MoviesComponent {...props}/>
      </StaticRouter>
    );
    
    expect(wrapper.find('MovieComponent').children().length).toEqual(2);
    expect(wrapper.find('footer').children().length).toEqual(1);
    expect(wrapper.find('.user-actions').children().length).toEqual(6);
  });

});