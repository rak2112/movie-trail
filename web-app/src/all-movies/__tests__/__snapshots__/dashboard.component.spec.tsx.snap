// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Component should match snapshot 1`] = `
<div
  className="css-5zmk01"
>
  <div
    className="css-182pldt"
  >
    <div
      className="Toastify"
    />
  </div>
  <div
    className="css-h5tpbd"
  >
    <div
      className="css-bxd89m"
    >
      <a
        href="/movie-details/2"
        onClick={[Function]}
      >
        <img
          alt=""
          src="http://image.tmdb.org/t/p/w500path "
        />
      </a>
      <div
        className="css-1lw9bxc"
      >
        <div
          className="user-actions"
        >
          <span
            className="rating"
          >
            vote_avg
            <span
              className="glyphicon glyphicon-star"
            />
          </span>
          <a
            className="remove-from-favorites"
            onClick={[Function]}
            title="Remove From Favorites"
          >
            <span
              className="glyphicon glyphicon-heart"
            />
          </a>
          <a
            className="add-to-watchlist"
            onClick={[Function]}
            title="Add to List"
          >
            <span
              className="glyphicon glyphicon-bookmark add-to-list"
            />
          </a>
        </div>
      </div>
      <div
        className="css-1w2z22x"
      >
        <h3>
          title
        </h3>
        <p>
          <a
            href="/movie-details/2"
            onClick={[Function]}
          >
            View Details
          </a>
        </p>
      </div>
    </div>
  </div>
  <div
    className="css-h5tpbd"
  >
    <div
      className="css-bxd89m"
    >
      <a
        href="/movie-details/3"
        onClick={[Function]}
      >
        <img
          alt=""
          src="http://image.tmdb.org/t/p/w500path "
        />
      </a>
      <div
        className="css-1lw9bxc"
      >
        <div
          className="user-actions"
        >
          <span
            className="rating"
          >
            vote_avg
            <span
              className="glyphicon glyphicon-star"
            />
          </span>
          <a
            className="add-to-favorites"
            onClick={[Function]}
            title="Add to Favorites"
          >
            <span
              className="glyphicon glyphicon-heart add-to-list"
            />
          </a>
          <a
            className="add-to-watchlist"
            onClick={[Function]}
            title="Add to List"
          >
            <span
              className="glyphicon glyphicon-bookmark add-to-list"
            />
          </a>
        </div>
      </div>
      <div
        className="css-1w2z22x"
      >
        <h3>
          title
        </h3>
        <p>
          <a
            href="/movie-details/3"
            onClick={[Function]}
          >
            View Details
          </a>
        </p>
      </div>
    </div>
  </div>
  <footer
    className="css-dpqy78"
  >
    <ul
      className="pagination"
    >
      <li
        className="disabled"
        onClick={[Function]}
      >
        <a
          className=""
          href="#"
        >
          «
        </a>
      </li>
      <li
        className="disabled"
        onClick={[Function]}
      >
        <a
          className=""
          href="#"
        >
          ⟨
        </a>
      </li>
      <li
        className="active"
        onClick={[Function]}
      >
        <a
          className="undefined"
          href="#"
        >
          1
        </a>
      </li>
      <li
        className="disabled"
        onClick={[Function]}
      >
        <a
          className=""
          href="#"
        >
          ⟩
        </a>
      </li>
      <li
        className="disabled"
        onClick={[Function]}
      >
        <a
          className=""
          href="#"
        >
          »
        </a>
      </li>
    </ul>
  </footer>
</div>
`;
